contractcourt/channel_arbitrator.go

```diff
@@ -509,6 +509,30 @@ func (c *ChannelArbitrator) stateStep(triggerHeight uint32,
 	// commitment transaction. We enter this state either due to an outside
 	// sub-system, or because an on-chain action has been triggered.
 	case StateBroadcastCommit:
+		// Under normal operation, we can only enter
+		// StateBroadcastCommit via a user or chain trigger. On restart,
+		// this state may be reexecuted after closing the channel, but
+		// failing to commit to StateContractClosed or
+		// StateFullyResolved. In that case, one of the three close
+		// triggers will be presented, signifying that we should skip
+		// rebroadcasting, and go straight to resolving the on-chain
+		// contract or marking the channel resolved.
+		switch trigger {
+		case localCloseTrigger, remoteCloseTrigger:
+			log.Infof("ChannelArbitrator(%v): detected %s "+
+				"close after closing channel, fast-forwarding "+
+				"to %s to resolve contract",
+				c.cfg.ChanPoint, trigger, StateContractClosed)
+			return StateContractClosed, closeTx, nil
```

Comment:
Maybe we should just change the structure of the whole method to just return the next state directly?

Reply:
I agree that’d be easier to follow :)

Reply:
I think the reason we didn't do that before was that we committed to the next state at the end. As we don't do that anymore, the change should make it read nicer I think :)

Reply:
Only other reason I can see would be to catch the log statement at the end, but that can be lifted outside of this function